server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/QuotaChangesMethod.scala

```diff
@@ -74,26 +69,15 @@ class QuotaChangesMethod @Inject()(val metricFactory: MetricFactory,
 
 }
 
-case class QuotaChangesResolver(private var quotaManager: QuotaManager,
-                                private var quotaRootResolver: UserQuotaRootResolver) {
-
+case class QuotaChangesResolver(private val quotaManager: QuotaManager,
+                                private val quotaRootResolver: UserQuotaRootResolver) {
   private val jmapQuotaManagerWrapper: JmapQuotaManagerWrapper = JmapQuotaManagerWrapper(quotaManager, quotaRootResolver)
 
-  def getLatestState(username: Username, capabilities: Set[CapabilityIdentifier]): SMono[State] =
-    jmapQuotaManagerWrapper.retrieveQuotaRoot(username, capabilities)
-      .flatMap(quotaRoot => SMono(quotaManager.getQuotasReactive(quotaRoot)))
+  def getLatestState(username: Username, capabilities: Set[CapabilityIdentifier]): SMono[(UuidState, Seq[Id])] =
+    jmapQuotaManagerWrapper.list(username, capabilities)
       .collectSeq()
-      .map(quotasList => evaluateStateFromQuotas(username, capabilities, quotasList))
-
-
-  private def evaluateStateFromQuotas(username: Username, capabilities: Set[CapabilityIdentifier], quotasList: Seq[QuotaManager.Quotas]): State = {
-    if (!quotasList.exists(quotas => quotas.getMessageQuota.getLimit.isLimited || quotas.getStorageQuota.getLimit.isLimited)) {
-      throw new CanNotCalculateChangesException("Quota is empty, hence we cannot calculate changes")
-    }
-
-    val usedTotal: Long = quotasList
-      .map(quotas => quotas.getMessageQuota.getUsed.asLong() + quotas.getStorageQuota.getUsed.asLong())
-      .sum
-    State.of(UUID.nameUUIDFromBytes(s"${username.toString}:${capabilities.map(_.value).mkString("_")}:${usedTotal}".getBytes(StandardCharsets.UTF_8)))
-  }
+      .doOnNext(quotas => if (quotas.isEmpty) {
+        throw new CanNotCalculateChangesException("Quota is empty, hence we cannot calculate changes")
+      })
```

Comment:
Why ?

Technically `JmapQuota.correspondingState(quotas)` handles empty quotas, that are stable. 

IMO drop this case.